src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemConsumer.cs

```diff
@@ -108,7 +106,14 @@ private async Task ReadAndProcessWorkItemAsync(QueueClient queueClient, WorkItem
             try
             {
                 _logger.LogInformation("Starting attempt {attemptNumber} for {workItemType}", message.DequeueCount, workItemType);
-                await workItemScope.RunWorkItemAsync(node, telemetryScope, cancellationToken);
+                await workItemScope.RunWorkItemAsync(
+                    node,
+                    telemetryScope,
+                    // We record the delay between the message being queued and the processing start time
+                    // We must only measure that once we actually start processing the work item which might mean waiting for lock
+                    // This means that VMR work items which lock over the local VMR will report the actual delay
```

Comment:
I don't think we need this line, we only have 1 codeflow work item processor, so we should never be waiting over the local VMR. Where we do need to wait is in non codeflow work items if we have multiple triggers of the same subscription, or when we trigger batched subscriptions

Reply:
True